tgg/relatedness/n_non_ref_pairs.py

```diff
@@ -0,0 +1,273 @@
+) -> hl.Table:
+    """
+    Filter VDS to autosomal sites in interval QC pass regions with an adj non ref of n.
+    :param vds_path: Path to VDS. Default is VDS_PATH.
+    :param temp_path: Path to bucket to store Table and other temporary data. Default is TEMP_PATH.
+    :param tranche_data: UKB tranche data (data source and data freeze number). Default is TRANCHE_DATA.
+    :param autosomes_only: Filter VDS to autosomes. Defaults to True.
+    :param snp_only: Filter VDS to bi-allelic SNPs. Defaults to False.
+    :param het_only: Filter to only het samples. Defaults to False.
+    :param adj_only: Filter GT to adj. Defaults to True.
+    :param interval_qc_regions: Filter to interval QC regions. Defaults to True.
+    :param no_AS_lowqual: Remove AS_lowqual sites. Defaults to True.
+    :param non_ref_samples: Number of non ref samples found in each variant to filter to, e.g. for tripletons, n_samples=3. Defaults to 3.
+    :return: Table of high quality sites with private to n samples.
+    """
+    vds = hl.vds.read_vds(vds_path)
+    mt = vds.variant_data
+
+    if autosomes_only:
+        logger.info("Filter to autosomes...")
+        mt = mt.filter_rows(mt.locus.in_autosome())
+
+    if snp_only:
+        logger.info("Filtering to bi-allelic SNPs...")
+        mt = mt.filter_rows(
+            bi_allelic_expr(mt) & hl.is_snp(mt.alleles[0], mt.alleles[1])
+        )
+
+    if no_AS_lowqual:
+        logger.info("Removing AS_lowqual sites...")
+        info_ht = hl.read_table(info_ht_path(*tranche_data, split=True))
+        mt = mt.filter_rows(~info_ht[mt.row_key].AS_lowqual)
+
+    if interval_qc_regions:
+        logger.info("Filtering to interval QC pass regions...")
+        interval_ht = hl.read_table(interval_qc_path(*tranche_data, "autosomes"))
+        mt = mt.filter_rows(hl.is_defined(interval_ht[mt.locus]))
+
+    if adj_only:
+        logger.info("Filtering to adj and calculating allele count...")
+        mt = filter_to_adj(mt)
+
+    logger.info("Changing old callstat annotation...")
+    mt = mt.transmute_rows(original_call_stats=mt.call_stats)
+
+    logger.info("Recalculating call stats post-quick QC...")
+    mt = mt.annotate_rows(hail_call_stats=hl.agg.call_stats(mt.GT, mt.alleles))
+
+    logger.info(
+        "Get AC at alt allele(call_stats includes a count for each allele, including reference) and annotate with hom_var, het, hom_alt..."
+    )
+    mt = mt.annotate_rows(
+        ac=mt.hail_call_stats.AC[1],
+        n_het=hl.agg.count_where(hl.is_defined(mt["GT"]))
+        - hl.sum(mt.hail_call_stats.homozygote_count),
+        n_hom_ref=mt.hail_call_stats.homozygote_count[0],
+        n_hom_alt=mt.hail_call_stats.homozygote_count[1],
+    )
+    mt = mt.annotate_rows(
+        n_non_ref=hl.agg.count_where(hl.is_defined(mt["GT"])) - mt.n_hom_ref
+    )
+
+    logger.info("Filtering sites to where n_non_ref = %i...", non_ref_samples)
+    mt = mt.filter_rows(mt.n_non_ref == non_ref_samples)
+
+    if het_only:
```

Comment:
just wondering, did you set this to default to False to rescue Mendelian violations (true variation or technical artifacts)? for example, if `n_non_ref` is 3, is this False to capture scenarios where one person is 1/1 and the other two are 0/1?

Reply:
> is this False to capture scenarios where one person is 1/1 and the other two are 0/1?
Yeah exactly. I brought this up in a standup last week. I wanted some clarity around what we were looking for/ "tripletons" since an AC=3 filter would remove sites that are like your example, private to three individuals but AC=4. Grace pointed out we should actually apply an n non ref filter which expands our AC threshold and thus this script. I added the het_only incase someone wanted to filter to the doubletons/AC=2.